### PR TITLE
MINOR: explain producer naming within Streams

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck @enothereska @dguy @guozhangwang 

Should go into `0.11`, too.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5486/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5471/
Test PASSed (JDK 8 and Scala 2.12).

File: docs/streams.html [hunk 1]
```diff
@@ -894,6 +894,19 @@
...
+            If exactly-once processing is enabled via parameter <code>processing.guarantees</code>, internally Streams switches from a producer per thread to a producer per task runtime model.
+            In order to distinguish the different producers, the producer's <code>client.id</code> additionally encodes the task-ID for this case.
+            Because the producer's <code>client.id</code> is used to report JMX metrics, it might be required to update tools that receive those metrics.
+        </p>
+        <p> Producer's <code>client.id</code> naming schema: </p>
+        <ul>
+            <li> at-least-once (default): <code>[client.Id]-StreamThread-[sequence-number]</code> </li>
+            <li> exactly-once: <code>[client.Id]-StreamThread-[sequence-number]-[taskId]</code> </li>
+        </ul>
+        <p> With <code>[client.Id]</code> is either set via Streams configuration parameter <code>client.id<code> or defaults to <code>[application.id]-[processId]</code> (<code>[processId]</code> is a random UUID). </p>
```
Comment (dguy) [inline docs/streams.html]: With -> The ?

File: docs/streams.html [hunk 2]
```diff
@@ -894,6 +894,19 @@
            If you want to view the changelog stream of the  <code>KTable</code> then you could call <code>KTable.toStream().print()</code>.
         </p>
 
+        <p> Metrics using exactly-once semantics: </p>
+        <p>
+            If exactly-once processing is enabled via parameter <code>processing.guarantees</code>, internally Streams switches from a producer per thread to a producer per task runtime model.
```
Comment (dguy) [inline docs/streams.html]: "If exactly-once processing is enabled via the <code>processing.gaurantees</code> parameter, ..."

**Comment by mjsax** [comment]
Updated this.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5515/
Test FAILed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5500/
Test PASSed (JDK 8 and Scala 2.12).

**Review by bbejeck — APPROVED** [review]

**Review by dguy — APPROVED** [review]
LGTM

**Comment by dguy** [comment]
Merged to trunk.

Sorry i did try to cherry-pick to 0.11.0 but it failed as i had a modified file checked-out

**Comment by mjsax** [comment]
@dguy Should I open a new PR against 0.11 ?

**Comment by dguy** [comment]
@mjsax - yes please

**Comment by mjsax** [comment]
Done: https://github.com/apache/kafka/pull/3401

**Pull request closed without merging**